src/docs/stan-reference/functions.tex

```diff
@@ -3424,6 +3424,127 @@ \chapter{Mixed Operations}\label{mixed-operations.chapter}
 %
 \end{description}
 
+\chapter{Algebraic Equation Solver}
+\label{functions-algebraic-solver.chapter}
+
+Stan provides a built-in algebraic equation solver. Although it 
+looks like other function applications, the algebraic solver is
+special in two ways.
+
+First, the algebraic solver is a higher-order function, i.e. it takes
+another function as one of its arguments. The only other functions
+in Stan which share this feature are the ordinary differential equation
+solvers (see section~\ref{functions-ode-solver.chapter}). Ordinary Stan
+functions do not allow functions as arguments.
+
+Second, some of the arguments of the algebraic solvers are restricted
+to data only expressions. These expressions must not contain variables
+other than those declared in the data or transformed data blocks. Ordinary
+Stan functions place no restriction on the origin of variables in their argument
+expressions.
+
+\section{Specifying an Algebraic Equation as a Function}
+\label{functions-algebra-function.section}
+
+An algebraic system is specified as an ordinary function in Stan within the
+function block. The algebraic system function must have this signature:
+%
+\begin{stancode}
+real[] algebra_system(real[] unknown, real[] theta,
+                      real[] x_r, int[] x_i)
+\end{stancode}
+%
+The algebraic system function should return the value of the algebraic
+function which goes to 0, when we plug in the solution to the algebraic
+system.
+
+The argument of this function are:
+\begin{itemize}
+\item \farg{unknown}, the unknowns we wish to solve for
+\item \farg{theta}, parameter values used to evaluate the algebraic system
+\item \farg{x\_r}, data values used to evaluate the algebraic system
+\item \farg{x\_i}, integer data used to evaluate the algebraic system
+\end{itemize}
+
+The algebraic system function separates parameter values, \farg{theta},
+from data values, \farg{x\_r}, for efficiency in computing the gradients
+of the algebraic system.
+
+\section{Call to the Algebraic Solver}
+
+\begin{description}
+%
+   \fitemthreelines{real[~]}{algebra\_solver}%
+                  {function \farg{algebra\_system}}%
+                  {real[] \farg{initial\_guess}}%
+                  {real \farg{theta}, real[] \farg{x\_r}, int[] \farg{x\_i}}%
+                  {Solves the algebraic system, given an initial guess,
+                    using the Powell Hybrid algorithm with the unsupported
+                    implementation from Eigen.}
```

Comment:
> I'd get rid of the " with the unsupported implementation from Eigen". It's in Stan now and we're supporting it.

My point is not to caution the function is unsupported, rather to state where the function we built the algebraic solver on comes from.